dbms/src/Common/getNumberOfPhysicalCPUCores.cpp

```diff
@@ -15,6 +17,22 @@ unsigned getNumberOfPhysicalCPUCores()
 {
 #if defined(__x86_64__)
 
+    std::ifstream cgroup_read_in("/sys/fs/cgroup/cpu/cpu.cfs_quota_us");
+    if (cgroup_read_in.is_open())
+    {
+        std::string allocated_cpus_share_str{ std::istreambuf_iterator<char>(cgroup_read_in), std::istreambuf_iterator<char>() };
+        int allocated_cpus_share_int = std::stoi(allocated_cpus_share_str);
+
+        cgroup_read_in.close();
+
+        // If a valid value is present
+        if(allocated_cpus_share_int > -1)
+        {
+            unsigned int allocated_cpus = round (allocated_cpus_share_int / 1000);
```

Comment:
Bug: round do nothing as you use integer division.

Reply:
Done